lib/stimulus_reflex/channel.rb

```diff
@@ -23,6 +23,7 @@ def receive(data)
     reflex_name, method_name = target.split("#")
     reflex_name = reflex_name.classify
     arguments = data["args"] || []
+    arguments.map! { |argument| map_hashes_in(argument) }
```

Comment:
Looks like `with_indifferent_access` will handle the nested values for us. Here's an example:

```ruby
{foo: 1, bar: [{baz: 2}]}.with_indifferent_access["bar"][0]["baz"] # => 2
```

I'm thinking we could simplify to this a one liner and get rid of the helper method.

```ruby
arguments.map! { |arg| arg.try(:with_indifferent_access) || nil }
```

Thoughts?

Reply:
@hopsoft Well, I was trying to be too clever by half by mapping incoming arrays that might have hashes inside of them instead of assuming each argument is just a hash… If we do go this direction with the arguments, I think we should at least make the attempt to check things in a recursive fashion somehow.

Reply:
Note that my solution doesn't assume every argument is a `Hash` because it's using ActiveSupport's `try` functionality. So if `with_indifferent_access` isn't supported by the argument, we simply return the arg itself.

The one use case this doesn't handle would be an array argument that contains Hashes. 🤔  I'm considering if we need to cover that use case.

Reply:
That's what I was trying to do, yeah. Hash arg is easy, array arg means looping through looking for hashes.

Reply:
Arrays containing hashes are reasonably common. *goes back to lurking*

Reply:
I'm sold. What about something like this:

```ruby
arguments.map! { |arg|
  arg.try(:with_indifferent_access) || arg.try(:map, &:with_indifferent_access) || arg
}
```

Too clever?